hypershift-operator/controllers/scheduler/autoscaler.go

```diff
@@ -27,22 +33,83 @@ import (
 )
 
 const (
-	autoscalerControllerName  = "RequestServingNodeAutoscaler"
-	descalerControllerName    = "MachineSetDescaler"
-	machineSetAnnotation      = "hypershift.openshift.io/machineset"
-	machineSetNamespace       = "openshift-machine-api"
-	machineNameNodeAnnotation = "machine.openshift.io/machine"
-	machineMachineSetLabel    = "machine.openshift.io/cluster-api-machineset"
+	autoscalerControllerName                  = "RequestServingNodeAutoscaler"
+	descalerControllerName                    = "MachineSetDescaler"
+	nonRequestServingAutoscalerControllerName = "NonRequestServingNodeAutoscaler"
+	machineSetAnnotation                      = "hypershift.openshift.io/machineset"
+	machineSetNamespace                       = "openshift-machine-api"
+	machineNameNodeAnnotation                 = "machine.openshift.io/machine"
+	machineMachineSetLabel                    = "machine.openshift.io/cluster-api-machineset"
+	clusterAPIMachineTypeLabel                = "machine.openshift.io/cluster-api-machine-type"
+	nonRequestServingLabelPrefix              = "non-serving"
+	maxSizeMachineSetAnnotation               = "machine.openshift.io/cluster-api-autoscaler-node-group-max-size"
+	minSizeMachineSetAnnotation               = "machine.openshift.io/cluster-api-autoscaler-node-group-min-size"
 )
 
 const (
 	// nodeScaleDownDelay is the amount of time a node must exist before it is considered for scaling down
 	nodeScaleDownDelay = 5 * time.Minute
-
-	// pendingWaitTime is the amount of time a pod must be pending before it is considered for scaling
-	pendingWaitTime = 1 * time.Minute
 )
 
+type NonRequestServingNodeAutoscaler struct {
+	client.Client
+}
+
+func (r *NonRequestServingNodeAutoscaler) SetupWithManager(mgr ctrl.Manager) error {
+	r.Client = mgr.GetClient()
+	builder := ctrl.NewControllerManagedBy(mgr).
+		For(&hyperv1.HostedCluster{}).
+		WithOptions(controller.Options{
+			RateLimiter:             workqueue.NewItemExponentialFailureRateLimiter(1*time.Second, 10*time.Second),
+			MaxConcurrentReconciles: 1,
+		}).
+		Named(nonRequestServingAutoscalerControllerName)
+	return builder.Complete(r)
+}
+
+func (r *NonRequestServingNodeAutoscaler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
If you consume `ClusterSizingConfiguration` and `MachineSet` we need to watch them as well so we react to changes to those independent of `HostedCluster`